src/providers/mod.rs

```diff
@@ -24,4 +24,44 @@ pub trait Provider {
     fn name(&self) -> &str;
     fn detect(&self, app: &App, _env: &Environment) -> Result<bool>;
     fn get_build_plan(&self, _app: &App, _environment: &Environment) -> Result<Option<BuildPlan>>;
+    fn get_metadata(&self, _app: &App, _env: &Environment) -> Result<ProviderMetadata> {
+        Ok(ProviderMetadata::default())
+    }
+}
+
+#[derive(Default)]
+pub struct ProviderMetadata {
+    pub labels: Option<Vec<String>>,
```

Comment:
Option<Vec is redundant, you should just use a Vec with nothing inside of it.

Reply:
`Option` is still useful if you ever want to deserialize data into a Rust struct since it will not error if the field is missing.